source/appModules/explorer.py

```diff
@@ -165,11 +165,22 @@ def _get_states(self):
 		states = super(StartButton, self).states
 		states.discard(controlTypes.STATE_SELECTED)
 		return states
+		
+CHAR_LTR_MARK = u'\u200E'
+CHAR_RTL_MARK = u'\u200F'
+class UIProperty(UIA):
+	#Used for columns in Windows Explorer Details view.
+	#These can contain dates that include unwanted left-to-right and right-to-left indicator characters.
+	
+	def _get_value(self):
+		value = super(UIProperty, self).value
+		return value.replace(CHAR_LTR_MARK,'').replace(CHAR_RTL_MARK,'')
 
 
 class AppModule(appModuleHandler.AppModule):
 
 	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
+		# Optimization: return early to avoid comparing class names and roles that will never match.

```

Comment:
The problem with this, is that we don't know which returns are purely optimisation, and which are functional, if any.  Perhaps there is no easy way to convey this information, and someone intending to modify this function will have to work it out. If there was a way that was not overly verbose, it would reduce the maintenance cost of this function.